runtime/near-vm-errors/src/lib.rs

```diff
@@ -1,93 +1,130 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_rpc_error_macro::RpcError;
+use serde::{Deserialize, Serialize};
 use std::fmt;
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(
+    Debug, Clone, PartialEq, Eq, BorshDeserialize, BorshSerialize, Deserialize, Serialize, RpcError,
+)]
+#[rpc_error_variant = "FunctionCall"]
 pub enum VMError {
-    FunctionCallError(FunctionCallError),
+    FunctionExecError(FunctionExecError),
+    // TODO: serialize/deserialize?
     StorageError(Vec<u8>),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub enum FunctionCallError {
+#[derive(
+    Debug, Clone, PartialEq, Eq, BorshDeserialize, BorshSerialize, Deserialize, Serialize, RpcError,
+)]
+#[rpc_error_variant = "FunctionExecError"]
+pub enum FunctionExecError {
     CompilationError(CompilationError),
-    LinkError(String),
+    LinkError { msg: String },
```

Comment:
why do we need `msg` as a field?

Reply:
We decided to use a named struct variants for errors, cause now runtime errors are serialized and passed to the client (errors now are a part of a public interface). This is a schema which is generated using a [`RpcError` derive macro](https://github.com/nearprotocol/nearcore/pull/1854/files/216c808a7acf73dcdb537c0f52d75a3d3eb73990#diff-19c78e09a686045ecce938c7023d794e): https://github.com/nearprotocol/nearcore/blob/216c808a7acf73dcdb537c0f52d75a3d3eb73990/res/rpc_errors_schema.json